### DISCUSSION TITLE
AzureSignTool with WildCards

### DISCUSSION THREAD

**Discussion by ghost** [vcsjones/AzureSignTool#130] [Q&A]
Hi,

I'm using your tool to sign my build output. Everything works great as long as I pass in a single file at a time.
What I wanted to know is how to use the command with a wildcard.

This is the script I'm using - the values are provided as parameters from the pipeline

```
[CmdletBinding()]
param (
    $SigningURL,
    $SigningVaultURL,
    $SigningClientId,
    $SigningClientSecret,
    $SigningCertName,
    $TimestampURL,
    $FileToSign
)

AzureSignTool.exe sign -du "$($SigningURL)" -kvu "$($SigningVaultURL)" -kvi "$($SigningClientId)" -kvs "$($SigningClientSecret)" -kvc "$($SigningCertName)" -tr "$($TimestampURL)" -debug -v "$($FileToSign)"
```

As a value for `$FileToSign` I wanted to provide smth like `**\*.exe`

Is this behavior supported out-of-the-box? If not I could wrap the call to this script and resolve the wildcards before passing it in to the AzureSignTool.

Let me know what you think

Regards
Norbert

**Comment by JulienMaille** [comment]
Have you figured it out?

**Reply by nhuethmayr** [reply]
Nope, I went with a PS script which iterates the files....

```
(Get-ChildItem -Path $SigningPath -Include $SigningFilePattern -Recurse) | ForEach-Object {
    Write("Found $($_)");

    & $PSScriptRoot\CodeSigning.ps1 -SigningURL $($SigningURL) -SigningVaultURL $($SigningVaultURL) -SigningClientId $($SigningClientId) -SigningClientSecret $($SigningClientSecret) -SigningCertName $($SigningCertName) -TimestampURL $($TimestampURL) -FileToSign $($_);
}
```
